### ISSUE TITLE
Assertion failure is a warning and does not display assertion position anymore?

### ISSUE THREAD

**Issue by denismerigoux** [CatalaLang/catala#720]
With `catala interpret`, assertion failure now displays a message like : 

```catala 
┌─[WARNING]─
│
│  Assertion failed: 582,92 € = 463,32 €
│
└─
```
However:  

* this message does not display which assertion has failed, which is really not helpful ;
* why is the assertion failure a warning and not an error that stops the computation?

**Comment by AltGr** [comment]
Related: https://github.com/CatalaLang/catala/commit/69e0a002dd31b95ec9bcd48ac1ac8d3503263be3#diff-f0c506b9a4056678c835ebde77dd6b585264c5613595517d311bbbb05713a8e1L809-L811

**Comment by denismerigoux** [comment]
For the `stop_on_error` flag, the assertions failures should be reported as errors, but the computation should continue, including potentially multiple assertion failures. But at the end, the result of the computation should not be shown (since assertion failed), and the exit code should be non-zero.

On top of the, we should add a specific flag `--no-fail-on-assert` where if activated (default false), transform assertion failures into warnings and let the computation continue normally.

**Comment by denismerigoux** [comment]
Closed by a2431ad
